app/models/settings.rb

```diff
@@ -0,0 +1,11 @@
+# frozen_string_literal: true
+
+require 'singleton'
+
+class Settings < ApplicationRecord
+  validates :singleton_guard, inclusion: { in: [0] }
+
+  def self.instance
+    first_or_create!(singleton_guard: 0)
+  end
+end
```

Comment:
```suggestion
class Settings < ApplicationRecord
  include Singleton
end
```

I think this is what you want

Reply:
Have you tried it?

In particular, the `Singleton` module makes `#allocate` private which prevents instantiating by way of loading from the database. That is, you can instantiate a new `Singleton` via `.instance` and even save it to the database, but you can't start a new Ruby process and retrieve that `Singleton from the database.

Reply:
I have a working demo on my machine. I sent it to Louis, I'm not sure when he'll push. I can push up a separate branch now if you really want to see it.

Reply:
Never mind, @frothedoatmilk  found that he can still create multiple `Settings` instances and the instance doesn't persist across server restarts, like you mentioned. I hadn't tested that. Sigh.

Reply:
It also still allows as many copies of Settings as you want through `Settings#create`

Reply:
I love when GitHub web doesn't show me comments and I repeat something someone has already said 🥰